### ISSUE TITLE
[Bug]: Permission.always.status returns permanentlyDenied even when can be requested

### ISSUE THREAD

**Issue by Smibser** [Baseflow/flutter-permission-handler#1145]
### Please check the following before submitting a new issue.

- [X] I have searched the [existing issues](https://github.com/baseflow/flutter-permission-handler/issues).
- [X] I have carefully [read the documentation](https://github.com/Baseflow/flutter-permission-handler/blob/main/permission_handler/README.md) and verified I have added the required platform specific configuration.

### Please select affected platform(s)

- [ ] Android
- [X] iOS
- [ ] Windows

### Steps to reproduce

1. request locationWhileInUse
2. Choose 'Allow While Using App' on the ios device
3. get status from location always permission

I already mentioned this issue here https://github.com/Baseflow/flutter-permission-handler/issues/1128. But since it's closed I decided to open another one. Maybe I am missing something again

### Expected results

the permission status of the always location should be denied since it never was requested but can be requested

### Actual results

the permission status is permantelyDenied. And therefore I have no way to check if I can request always permission or not

### Code sample

<details><summary>Code sample</summary>

```dart
    final locationWhenInUseStatus = await Permission.locationWhenInUse.request();
    if (locationWhenInUseStatus.isPermantlyDenied) {
      print("Can't request always permission");
    } else {
      var locationAlwaysStatus = await Permission.locationAlways.request();
      // is never reached since 'Allow Only Once' and 'Allow While Using App' both return permantlyDenied
    }
```

</details>

### Version

10.2.0

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.7.12, on macOS 13.3.1 22E772610a darwin-arm64, locale en-DE)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.2)
[✓] IntelliJ IDEA Ultimate Edition (version 2020.3.2)
[✓] VS Code (version 1.81.1)
[✓] Connected device (4 available)
[✓] HTTP Host Availability

• No issues found!
```

</details>

**Comment by JeroenWeener** [comment]
Hi @Smibser thank you for opening a new issue.

I did some investigation and this does indeed look like a bug which we should investigate.

Recording of the bug in action:
https://github.com/Baseflow/flutter-permission-handler/assets/9049645/a5622730-fbe8-4261-a934-a2fc8d8cf439

**Comment by jeffscaturro-aka** [comment]
I came across something that may be related to this bug as well, around `locationAlways` (on iOS).

It seems that `await Permission.locationAlways.request();` doesn't actually wait for the user's response, and returns right away (as the ask dialog is presented to the user, not as the user responds to the dialog).

```
  Future<bool> _requestLocationWhileInUse() async {
    // This actually waits for user to respond to dialog prompt before
    // continuing to return the status.isGranted.
    final status = await Permission.locationWhenInUse.request();

    return status.isGranted;
  }

  Future<bool> _requestLocationAlways() async {
    // This does not wait for user to respond to dialog prompt before
    // continuing to return the status.isGranted, thus, we often receive
    // false here, and have to check after the user responds to the dialog.
    final status = await Permission.locationAlways.request();

    return status.isGranted;
  }
```

**Comment by MendleM** [comment]
I'm experiencing this too but for `Permission.storage.request()`

**Comment by ly75x12** [comment]
I encountered the same problem when using `PermissionStatus permissionStatus = await Permission.photosAddOnly.request();`

Is there any solution?

**Comment by Justincyz** [comment]
same issue on Permission.bluetooth

**Comment by merliandrea** [comment]
Same problem with `Permission.photosAddOnly.request()`

**Comment by nayanAubie** [comment]
@Smibser The same issue I am facing but in a different way.
- Grant both permissions `While using the app` & `Always Allow`
- Then change the permission by Settings app from `Always` to `Ask Next Time or When I Share`
- After that, I can see a permission pop-up of `While using the app` but not for `Always allow`

**Comment by FFelipeWNeo** [comment]
Same problem with `Permission.photos.request()`, too.
Has anyone found a solution?

**Comment by mateendev3** [comment]
Actually for iOS if you are using PermissionHandler for permission handling. You need to specify the permissions in the info.plist file as well as in podFile.

Here is an example of podFile
`post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)

    target.build_configurations.each do |config|
      # You can remove unused permissions here
      # for more information: https://github.com/BaseflowIT/flutter-permission-handler/blob/master/permission_handler/ios/Classes/PermissionHandlerEnums.h
      # e.g. when you don't need camera permission, just add 'PERMISSION_CAMERA=0'
      config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= [
        '$(inherited)',

        ## dart: PermissionGroup.calendar
        'PERMISSION_EVENTS=1',
        
        ## dart: PermissionGroup.calendarFullAccess
        'PERMISSION_EVENTS_FULL_ACCESS=1',

        ## dart: PermissionGroup.reminders
        'PERMISSION_REMINDERS=1',

        ## dart: PermissionGroup.contacts
        'PERMISSION_CONTACTS=1',

        ## dart: PermissionGroup.camera
        'PERMISSION_CAMERA=1',

        ## dart: PermissionGroup.microphone
        'PERMISSION_MICROPHONE=1',

        ## dart: PermissionGroup.speech
        'PERMISSION_SPEECH_RECOGNIZER=1',

        ## dart: PermissionGroup.photos
        'PERMISSION_PHOTOS=1',

        ## The 'PERMISSION_LOCATION' macro enables the `locationWhenInUse` and `locationAlways` permission. If
        ## the application only requires `locationWhenInUse`, only specify the `PERMISSION_LOCATION_WHENINUSE`
        ## macro.
        ##
        ## dart: [PermissionGroup.location, PermissionGroup.locationAlways, PermissionGroup.locationWhenInUse]
        'PERMISSION_LOCATION=1',
        'PERMISSION_LOCATION_WHENINUSE=0',

        ## dart: PermissionGroup.notification
        'PERMISSION_NOTIFICATIONS=1',

        ## dart: PermissionGroup.mediaLibrary
        'PERMISSION_MEDIA_LIBRARY=1',

        ## dart: PermissionGroup.sensors
        'PERMISSION_SENSORS=1',

        ## dart: PermissionGroup.bluetooth
        'PERMISSION_BLUETOOTH=1',

        ## dart: PermissionGroup.appTrackingTransparency
        'PERMISSION_APP_TRACKING_TRANSPARENCY=1',

        ## dart: PermissionGroup.criticalAlerts
        'PERMISSION_CRITICAL_ALERTS=1',

        ## dart: PermissionGroup.criticalAlerts
        'PERMISSION_ASSISTANT=1',
      ]

    end
  end
end`

**Comment by icecandy** [comment]
I've been stuck on microphone permissions using Flutter on iOS for a few hours now.
Little did I know that you have to add the access configuration in the iPod file for it to work.
Why in gods name does it not state this in the ReadMe? It does explain about adding permissions in Info.plist but nothing about the iPod file.
